### ISSUE TITLE
Option to skip Data Classification, set default Classification and set default Visibility

### ISSUE THREAD

**Issue by wapster182** [pnp/provision-assist-m365#59]
### Suggestion

We would love a quick option within the List "Provisioning Request Setting" with the following options:

- Option to skip data classification (step 3)
- Set default data classification
- Set default visibility (e.g. always private for Teams)

**Comment by alexc-MSFT** [comment]
@wapster182 I am working on some of these currently. So far I have item 1 from your list above ticked off.

The easiest way for now to implement the remainder would be to have a default data classification and visibility across the board in the settings list i.e. not for each space type e.g. team, what do you think?

**Comment by alexc-MSFT** [comment]
Also with this one 'Set default data classification', do you mean setting default confidential data to Yes/No?

**Comment by wapster182** [comment]
Glad to hear!

Yes, that would be fine to have one default data classification/visibility for all space types.

Correct, we would like to skip the classification step and therefore set up a default one that will be set in the request (because I guess the value needs to be filled). So it would be fine to default select Yes/No (or confedential/public).

**Comment by alexc-MSFT** [comment]
Thanks @wapster182, just to explain how I've set this up to work (should match your thoughts above) before I realise it:

1. New setting in settings list to hide the data classification screen
2. New columns in the Provisioning Types list (see screenshot below) so you can set a default visibility and confidential data value
3. When a user selects the type in the recommendation screen the default values will be applied (irrespective of whether the screen is hidden)
4. If a user changes one of the values the changed value is kept until they start a new request (does not change if they change the space type)

![image](https://github.com/pnp/provision-assist-m365/assets/12395485/89b7622c-0aab-4657-a5a0-ae6751ae7191)

![image](https://github.com/pnp/provision-assist-m365/assets/12395485/54a8443c-7901-479b-b687-640f3ca0f25a)

Let me know your thoughts - thanks.

**Comment by wapster182** [comment]
Perfect, seems great to me. We would love to have it realized that way.

**Comment by alexc-MSFT** [comment]
@wapster182 ok this is now done, you will need to add the following setting to the settings list:

![image](https://github.com/pnp/provision-assist-m365/assets/12395485/963c2b04-966c-470d-a18c-bfa3016018ab)

Also, create the following columns in the Provisioning Types list please:

![image](https://github.com/pnp/provision-assist-m365/assets/12395485/148f70f1-0f2d-4d79-8488-7bf7741ed757)

Ensure the internal name of those columns **do not have any spaces**

**Comment by wapster182** [comment]
After importing the new Power App solution, the Data Classification Screen throws 3 errors:

![image](https://github.com/pnp/provision-assist-m365/assets/68082620/97b97f6d-0b3b-4a0f-986c-b69a9c9a4a98)

**Comment by wapster182** [comment]
It seems to me that the 'Default Visibility' column is not recognized. I relinked the data sources to the enviroment variables (as described in the deployment guide) and also refreshed them, still no success.

**Comment by wapster182** [comment]
Update: I digged deeper and it seems the .Value is the problem. When removing it, no error is shown anymore. So it seems the column is correctly loaded, but the data type is already text and using .Value is not possible there.

**Comment by wapster182** [comment]
Update2: I removed the .Value from the commands that were throwing errors and saved & published.
Loaded the App, the steps are not displayed anymore so seems to be working.

Could you please confirm that the .Value just needs to be removed and the type text is correct? I would then proceed with further testing of the default data classification and visibility :)

**Comment by alexc-MSFT** [comment]
@wapster182 sorry, my bad.

Default Visibility is a CHOICE column with the values - Private, Public

A text column will give you that error.

Just chance to choice and revert your changes.

**Comment by wapster182** [comment]
Makes sense, I thought it was a text column. Works great now! Can be closed.
